dynawo/sources/Simulation/DYNSimulation.cpp

```diff
@@ -652,6 +698,40 @@ Simulation::importCurvesRequest() {
   curvesCollection_ = CurvesCollectionFactory::copyInstance(importer.importFromFile(curvesInputFile_));
 }
 
+void
+Simulation::importFinalStateValuesRequest() {
+  // Obtain Final State Values definitions from input file
+  finalStateValues::XmlImporter importer;
+  boost::shared_ptr<FinalStateValuesCollection> finalStateValuesCollection =
+    FinalStateValuesCollectionFactory::copyInstance(importer.importFromFile(finalStateValuesInputFile_));
+
+  // Final State Values and Curve points will be recorded during simulation in the same data structures
+  // Here we merge the Final State Values definitions with the existing Curves definitions
+
+  // A map for existing Curves is built so we can locate them fast and update them from the Final State Values
+  // A Curve is identified by the pair model name, variable name
+  typedef boost::unordered_map<std::pair<std::string, std::string>, boost::shared_ptr<curves::Curve> > CurvesMap;
+  CurvesMap curvesMap;
+  for (CurvesCollection::const_iterator itCurve = curvesCollection_->cbegin(); itCurve != curvesCollection_->cend(); ++itCurve) {
+    curvesMap.insert({std::make_pair((*itCurve)->getModelName(), (*itCurve)->getVariable()), *itCurve});
+  }
+
+  for (FinalStateValuesCollection::const_iterator itFinalStateValue = finalStateValuesCollection->cbegin();
+      itFinalStateValue != finalStateValuesCollection->cend();
+      ++itFinalStateValue) {
+    CurvesMap::const_iterator entry = curvesMap.find(std::make_pair((*itFinalStateValue)->getModelName(), (*itFinalStateValue)->getVariable()));
+    if (entry != curvesMap.end()) {
+      entry->second->setExportType(curves::Curve::EXPORT_AS_BOTH);
+    } else {
+      boost::shared_ptr<curves::Curve> curve = curves::CurveFactory::newCurve();
+      curve->setModelName((*itFinalStateValue)->getModelName());
+      curve->setVariable((*itFinalStateValue)->getVariable());
+      curve->setExportType(curves::Curve::EXPORT_AS_FINAL_STATE_VALUE);
+      curvesCollection_->add(curve);
```

Comment:
It means that every time we dump the curves (so at each time step) we go through final state values too (even if nothing is stored). Cannot we have a second CurveCollection objects just for final states to avoid this?

Reply:
Curve entries corresponding to final state values will be updated wit the values at current step. We decided to use the same (unique) collection for simplicity. Also, an entry could be requested as a curve and as a final state value, we will have only one entry in the collection, and it will be used for the two kinds of output.

Reply:
Ok, but performance wise it means that we copy a value at the N timesteps whereas we would only need to copy it at the end if we are in final state mode

Reply:
Yes, from a performance point of view it would be better to set final state values only after simulation has finished

Reply:
OK, let's address this in another PR